src/bio.c

```diff
@@ -202,8 +202,9 @@ int wolfSSL_BIO_read(WOLFSSL_BIO* bio, void* buf, int len)
         }
     }
 
-    /* start at end of list and work backwards */
-    while ((bio != NULL) && (bio->next != NULL)) {
+    /* start at end of list (or a WOLFSSL_BIO_SSL object since it takes care of
```

Comment:
How does WOLFSSL_BIO_SSL object take care of the rest of the chain? Looking at wolfSSL_BIO_SSL_read it handles a single BIO that it is currently on, and performs wolfSSL_read() using it. But does not look at the next or previous BIO.

Reply:
When a BIO chain is created in `wolfSSL_BIO_push` then the next element in the chain is set as the IO for the SSL object with `wolfSSL_set_bio`. Then `BioReceive` and `BioSend` call `wolfSSL_BIO_read` and `wolfSSL_BIO_write` respectively on the rest of the chain.

Reply:
I see. There is also a change in this PR so that wolfSSL_BIO_push always calls wolfSSL_set_bio. Users could call wolfSSL_set_bio themselves after pushing an SSL BIO to a chain and break it ....but that may just be a use case that is not done. The behavior implemented in the PR seems to be similar to what OpenSSL is doing.